airflow/www/views.py

```diff
@@ -1724,6 +1742,68 @@ def gantt(self):
             root=root,
         )
 
+    @expose('/extra_links')
+    @login_required
+    @wwwutils.action_logging
+    def extra_links(self):
+        """
+        A restful endpoint that returns external links for a given Operator
+
+        It queries the operator that sent the request for the links it wishes
+        to provide for a given external link name.
+
+        API: GET
+        Args: dag_id: The id of the dag containing the task in question
+              task_id: The id of the task in question
+              execution_date: The date of execution of the task
+              redirect_to: The name of the link reference to find the actual URL for
+
+        Returns:
+            200: {url: <url of link>, error: None} - returned when there was no problem
+                finding the URL
+            404: {url: None, error: <error message>} - returned when the operator does
+                not return a URL
+        """
+        dag_id = request.args.get('dag_id')
+        task_id = request.args.get('task_id')
+        execution_date = request.args.get('execution_date')
+        redirect_to = request.args.get('redirect_to')
+        dttm = pendulum.parse(execution_date)
```

Comment:
I wouldn't introduce a new lib for this purpose, other endpoints should contain the same logic already

Reply:
Pendulum is used upstream to manage this stuff. We don't have it locally yet. I can switch it to a current library

Reply:
I think we could stick with Pendulum as upstream uses it for converting all the timzone info to utc.